protobuf/source_reflection_test.go

```diff
@@ -0,0 +1,74 @@
+package protobuf
+
+import (
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+func TestReflection(t *testing.T) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	defer ln.Close()
```

Comment:
nit: I'm not sure that we should assume the result is non-nil here. Should we instead re-order this to:
```
require.NoError(t, err)
defer ln.Close()
```